Assets/Scripts/SongSystem/SongKeyIndicate.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace XylophoneHero
+{
+    [RequireComponent(typeof(Renderer))]
+    [RequireComponent(typeof(AudioSource))]
+    public class SongKeyIndicate : MonoBehaviour
+    {
+
+        #region PUBLIC_MEMBER_VARIABLES
+
+        public KeyCode TestKeyCode;
+
+        public float HighlightTime = 0.1f;
+
+        #endregion // PUBLIC_MEMBER_VARIABLES
+
+        #region PRIVATE_MEMBER_VARIABLES
+
+        private GameObject m_SongNoteObject = null;
+        private Renderer m_Rend;
+        private AudioSource m_KeySound;
+
+        private Color m_OriginColor;
+        private Color m_CorrectColor = new Color(1.0f, 1.0f, 1.0f);
+        private Color m_ErrorColor = new Color(0.8f, 0.2f, 0.2f);
+        private const float m_Lerp = 0.6f;
+
+        private SongManager m_SongManager;
+
+        #endregion // PRIVATE_MEMBER_VARIABLES
+
+        #region MONOBEHAVIOR_METHODS
+        void Start()
+        {
+            m_Rend = GetComponent<Renderer>();
+            m_KeySound = GetComponent<AudioSource>();
+            m_OriginColor = m_Rend.material.GetColor("_Color");
+
+            m_SongManager = transform.parent.parent.Find("SongSystemManager").GetComponent<SongManager>();
```

Comment:
Make SongSystemManager a Singleton instead. The structure of the object could change in the future.